lib/node_modules/@stdlib/number/float16/base/ulp-difference/lib/main.js

```diff
@@ -0,0 +1,105 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var SIGN_MASK = require( '@stdlib/constants/float16/sign-mask' );
+var UINT16_MAX = require( '@stdlib/constants/uint16/max' );
+var toWord = require( '@stdlib/number/float16/base/to-word' );
+var abs = require( '@stdlib/math/base/special/abs' );
+
+
+// FUNCTIONS //
+
+/**
+* Converts an unsigned 32-bit integer corresponding to the IEEE 754 binary representation of a half-precision floating-point number to a lexicographically ordered integer.
+*
+* @private
+* @param {unsigned32} word - unsigned 32-bit integer
+* @returns {integer} lexicographically ordered integer
+*/
+function monotoneKey( word ) {
+	if ( word & SIGN_MASK ) { // x < 0
+		return ( ~word + 1 ) &UINT16_MAX; // two's-complement negation
```

Comment:
I am guessing that you are AND'ing the bits because you've done `~word` negation and because JS only has 32-bit integer ops, the leading zeros all get flipped on during negation and we want them to remain off. Am I correct?

The question I have, however, is whether two's complement negation actually works here, as the `+1` no longer causes the `word` to overflow as a 32-bit integer does. Have you actually verified that the negation logic works as you expect?